Mathlib/LinearAlgebra/Determinant.lean

```diff
@@ -486,6 +486,38 @@ theorem LinearEquiv.coe_ofIsUnitDet {f : M →ₗ[R] M'} {v : Basis ι R M} {v'
   ext x
   rfl
 
+/-- Builds a linear equivalence from an endomorphism whose determinant is a unit. -/
+noncomputable def LinearMap.equivOfIsUnitDet
+    [Module.Free R M] [Module.Finite R M]
+    {f : M →ₗ[R] M} (h : IsUnit f.det) :
+    M ≃ₗ[R] M := by
+  by_cases hR : Nontrivial R
+  · let ⟨ι, b⟩ := (Module.Free.exists_basis R M).some
+    have : Finite ι := Module.Finite.finite_basis b
+    have : Fintype ι := Fintype.ofFinite ι
+    have : DecidableEq ι := Classical.typeDecidableEq ι
+    exact LinearEquiv.ofIsUnitDet (by rwa [det_toMatrix b])
+  · exact 1
+
+@[simp]
+theorem LinearMap.equivOfIsUnitDet_apply
+    [Module.Free R M] [Module.Finite R M]
+    {f : M →ₗ[R] M} (h : IsUnit f.det) (x : M) :
+    (LinearMap.equivOfIsUnitDet h) x = f x := by
+  by_cases hR : Nontrivial R
+  · simp [equivOfIsUnitDet, dif_pos hR]
+  · rw [not_nontrivial_iff_subsingleton] at hR
+    have : Subsingleton M := Module.subsingleton R M
+    apply Subsingleton.allEq
```

Comment:
```suggestion
@[simp]
theorem LinearMap.equivOfIsUnitDet_apply
    [Module.Free R M] [Module.Finite R M]
    {f : M →ₗ[R] M} (h : IsUnit f.det) (x : M) :
    (LinearMap.equivOfIsUnitDet h) x = f x := by
  nontriviality M
  have hR : Nontrivial R := by
    by_contra! h
    have : Subsingleton M := Module.subsingleton R M
    exact false_of_nontrivial_of_subsingleton M
  simp [equivOfIsUnitDet, dif_pos hR]
```
The part saying that `R` is nontrivial should be extracted as a sublemma.

I tried to make `nontriviality` able to prove it but I didn't succeed.